test/lib/jdk/test/lib/Utils.java

```diff
@@ -818,4 +834,200 @@ public static Path createTempDirectory(String prefix, FileAttribute<?>... attrs)
+            while ((line = reader.readLine()) != null) {
+                sb.append(contentMapper.apply(line))
+                        .append(lineSep);
+            }
+        }
+
+        try (FileWriter writer = new FileWriter(source.toFile())) {
+            writer.write(sb.toString());
+        }
+    }
+
+    /**
+     * Replace files' string by applying the given mapper function.
+     *
+     * @param source        the file to read
+     * @param contentMapper the mapper function applied to files' content
+     * @throws IOException  if an I/O error occurs
+     */
+    public static void replaceFilesString(List<Path> source,
+            Function<String, String> contentMapper) throws IOException {
+        for (Path file : source) {
+            replaceFileString(file, contentMapper);
+        }
+    }
+
+    /**
+     * Grant file user access or full access to everyone.
+     *
+     * @param file   file to grant access
+     * @param userOnly true for user access, otherwise full access to everyone
+     * @throws IOException if error occurs
+     */
+    public static void grantFileAccess(Path file, boolean userOnly)
+            throws IOException {
+        Set<String> attr = file.getFileSystem().supportedFileAttributeViews();
+        if (attr.contains("posix")) {
+            if (userOnly) {
+                Files.setPosixFilePermissions(file,
+                        PosixFilePermissions.fromString("rwx------"));
+            } else { /* Allow access to everyone */
+                Files.setPosixFilePermissions(file,
+                        PosixFilePermissions.fromString("rwxrwxrwx"));
+            }
+        } else if (attr.contains("acl")) {
+            AclFileAttributeView view =
+                    Files.getFileAttributeView(file, AclFileAttributeView.class);
+            List<AclEntry> acl = new ArrayList<>();
+            for (AclEntry thisEntry : view.getAcl()) {
+                if (userOnly) {
+                    if (thisEntry.principal().getName()
+                            .equals(view.getOwner().getName())) {
+                        acl.add(AclEntry.newBuilder(thisEntry)
+                                .setType(AclEntryType.ALLOW)
+                                .build());
+                    } else if (thisEntry.type() == AclEntryType.ALLOW) {
+                        acl.add(AclEntry.newBuilder(thisEntry)
+                                .setType(AclEntryType.DENY)
+                                .build());
+                    } else {
+                        acl.add(thisEntry);
+                    }
+                } else {
+                    if (thisEntry.type() != AclEntryType.ALLOW) {
+                        acl.add(AclEntry.newBuilder(thisEntry)
+                                .setType(AclEntryType.ALLOW)
+                                .build());
+                    } else {
+                        acl.add(thisEntry);
+                    }
+                }
+            }
```

Comment:
What was your motivation to get rid of the Util methods revokeAccess, allowAccess and buildAclEntry.
The implementation did not become more clearer nor simpler. But reviewing it became much harder.

Reply:
Sure, I will add those methods back.